common/remote_object.go

```diff
@@ -40,17 +39,62 @@ func (pe *objectPropertyParseError) Unwrap() error {
 	return pe.error
 }
 
+type multiError struct {
+	Errors []error
+}
+
+func (me *multiError) append(err error) {
+	me.Errors = append(me.Errors, err)
+}
+
+func (me multiError) Error() string {
+	if len(me.Errors) == 0 {
+		return ""
+	}
+	if len(me.Errors) == 1 {
+		return me.Errors[0].Error()
+	}
+
+	var buf strings.Builder
+	buf.WriteString(fmt.Sprintf("%d errors occurred:\n", len(me.Errors)))
+	for _, e := range me.Errors {
+		buf.WriteString(fmt.Sprintf("\t* %s\n", e.Error()))
+	}
```

Comment:
You could use `fmt.Fprintf` here instead:

```suggestion
	fmt.Fprintf(&buf, "%d errors occurred:\n", len(me.Errors))
	for _, e := range me.Errors {
		fmt.Fprintf(&buf, "\t* %s\n", e)
	}
```

Reply:
Out of curiosity, why would you chose one over the other?

Reply:
Yes, it's more performant and less code (=easier to read and maintain). More function calls often result in a less performant code.

Reply:
You are right! Done.